### ISSUE TITLE
AttachVolume.FindAttachablePluginBySpec failed for volume

### ISSUE THREAD

**Issue by survivant** [openebs/cstor-operators#258]
I have pods that are stalled in init state or one was in 0/1 Running state.. but they all have the event : 
attachdetach-controller  AttachVolume.FindAttachablePluginBySpec failed for volume "xxxx"

````
Events:
  Type     Reason              Age                  From                     Message
  ----     ------              ----                 ----                     -------
  Warning  FailedAttachVolume  3m8s (x279 over 9h)  attachdetach-controller  AttachVolume.FindAttachablePluginBySpec failed for volume "pvc-21d65a88-5c98-4759-985c-e7af088798ff"
root@test-pcl109:~#
root@test-pcl109:~# kubectl get pv pvc-21d65a88-5c98-4759-985c-e7af088798ff
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                               STORAGECLASS    REASON   AGE
pvc-21d65a88-5c98-4759-985c-e7af088798ff   2Gi        RWO            Delete           Bound    default/datadir-twin-neo4j-core-0   sc-iep-mirror            4d17h
root@test-pcl109:~# kubectl -n openebs get cva | grep pvc-21d65a88-5c98-4759-985c-e7af088798ff
pvc-21d65a88-5c98-4759-985c-e7af088798ff-test-pcl111   4d17h
root@test-pcl109:~# kubectl -n openebs get cvr | grep pvc-21d65a88-5c98-4759-985c-e7af088798ff
pvc-21d65a88-5c98-4759-985c-e7af088798ff-cspc-iep-mirror-zqbr    2.09M       17.3M   Healthy   4d17h
root@test-pcl109:~# kubectl -n openebs get cvc | grep pvc-21d65a88-5c98-4759-985c-e7af088798ff
pvc-21d65a88-5c98-4759-985c-e7af088798ff   2Gi        Bound    4d17h
root@test-pcl109:~# kubectl -n openebs get pods | grep pvc-21d65a88-5c98-4759-985c-e7af088798ff
pvc-21d65a88-5c98-4759-985c-e7af088798ff-target-5dfd4448ccrbv5f   3/3     Running   0          41h
root@test-pcl109:~#
````

````
1. kubectl get csidriver cstor.csi.openebs.io
2. kubectl get sts openebs-cstor-csi-controller -n openebs -oyaml


root@test-pcl109:~# kubectl get csidriver cstor.csi.openebs.io
NAME                   ATTACHREQUIRED   PODINFOONMOUNT   MODES                  AGE
cstor.csi.openebs.io   false            true             Persistent,Ephemeral   5d22h
root@test-pcl109:~#

````
[controler-log.txt](https://github.com/openebs/cstor-operators/files/6062974/controler-log.txt)

I deleted that pod and it returned to RUNNING STATE fine.   BUT after that.  I got other pods that were previously in "Crashloop" because they were waiting after the database to came online (the pod that I previously deleted).

Some of them came back online fine.. but I have other pods that got the error : 
````
AttachVolume.FindAttachablePluginBySpec failed for volume
````

I created a new cluster last week with kubeadm.  And reinstalled OpenEBS cstor from ctor Helm chart.  and installed my application after recreating the cstor pool.

````
NAME          STATUS   ROLES    AGE     VERSION
test-pcl109   Ready    master   6d23h   v1.18.4
test-pcl110   Ready    <none>   6d23h   v1.18.4
test-pcl111   Ready    <none>   6d23h   v1.18.4
test-pcl112   Ready    <none>   6d23h   v1.18.4
test-pcl113   Ready    <none>   6d23h   v1.18.4
root@test-pcl109:~#
````

from OpenEBS team in Slack :  Since our new version of csidiver doesn't support attach, attach should not be tried by kubernetes. I think kubernetes is looking at some cashed version of the driver when the error occurred (Not sure). We have disabled attach/detach functionality in latest cstor-csi versions (2.6.0).

**Comment by survivant** [comment]
I was asked to provide those logs in Slack.  The problem was not present anymore because I uninstalled my applications and reinstalled them.  BUT I got the error later this morning.. same message : AttachVolume.FindAttachablePluginBySpec failed for volume "xxxx" 

[node113.txt](https://github.com/openebs/cstor-operators/files/6069845/node113.txt)
[node110.txt](https://github.com/openebs/cstor-operators/files/6069846/node110.txt)
[node111.txt](https://github.com/openebs/cstor-operators/files/6069847/node111.txt)
[node112.txt](https://github.com/openebs/cstor-operators/files/6069848/node112.txt)

**Comment by payes** [comment]
hi @survivant, I am looking into this.

**Comment by survivant** [comment]
thanks.

**Comment by payes** [comment]
This issue is related to kubernetes https://github.com/kubernetes/kubernetes/issues/89173. It has been fixed in kubernetes 1.20+.
Ref: https://github.com/kubernetes/kubernetes/commit/e23d83eead3b5ae57731afb0209f4a2aaa4009dd
We have tried this with kubernetes 1.18.4, 1.19.5 and 1.20.2, it was hit in kubernetes 1.18 and 1.19 but 1.20 works fine.
Steps followed: https://github.com/kubernetes/kubernetes/issues/89173#issue-583009942
